src/ui/components/ide/folder/folder.tsx

```diff
@@ -294,6 +294,7 @@ export function Folder(props: {
 					getFolderContextMenu(props.path, {
 						onAddNewClick() {
 							setIsBeingAddedTo(true);
+							setOpened((open) => !open); // change folder open state when adding new file
```

Comment:
Let's be explicit about the state of 'opened' here

Reply:
by being explicit, do you mean to set the state directly here instead on depending on prevState ? In that case the solution would be just set the opened state as false `setOpened(false);`. should I raise a new pr for this ?

Reply:
That sounds workable but the end result would look like:
- open folder with lot of files under it
- right click
- select 'Add new'
- and the the folder will fully collapse

This can be a bit jarring UX. Can we figure out what the root cause it and fix it in such a way that opening or closing a folder has no impact on adding new files?